src/test/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundMessageHandlerTest.java

```diff
@@ -47,56 +60,110 @@
 
     @BeforeEach
     void setUp() {
-        when(interchange.getInterchangeHeader()).thenReturn(new InterchangeHeader());
+        lenient().when(interchange.getInterchangeHeader()).thenReturn(
+            InterchangeHeader.builder()
+                .sender("some_sender")
+                .recipient("some_recipient")
+                .sequenceNumber(SEQUENCE_NUMBER)
+                .translationTime(Instant.now())
+                .build());
     }
 
     @Test
-    void handleInboundMeshMessageNoMessagesDoesNotPublishToGpOutboundQueue() {
+    void handleInvalidInterchangeMeshMessageRaisesException()
+            throws InterchangeParsingException, MessagesParsingException {
 
-        final MeshMessage meshMessage = new MeshMessage();
+        final MeshMessage meshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY);
+        when(edifactParser.parse(meshMessage.getContent())).thenThrow(InterchangeParsingException.class);
+
+        final OutboundMeshMessage outboundMeshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY_ACK);
+        when(outboundMeshMessageBuilder.buildNhsAck(eq(WorkflowId.PATHOLOGY), isA(InterchangeParsingException.class)))
+            .thenReturn(outboundMeshMessage);
 
+        inboundMessageHandler.handle(meshMessage);
+
+        verify(edifactParser, times(1)).parse(any());
+        verify(edifactToFhirService, never()).convertToFhir(any(Message.class));
+        verify(gpOutboundQueueService, never()).publish(any(Bundle.class));
+        verify(outboundMeshMessageBuilder, never()).buildNhsAck(any(), any(), anyList());
+        verify(meshOutboundQueueService).publish(eq(outboundMeshMessage));
+    }
+
+    @Test
+    void handleInvalidMessageMeshMessageRaisesException() throws InterchangeParsingException, MessagesParsingException {
+        final MeshMessage meshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY);
+        when(edifactParser.parse(meshMessage.getContent())).thenThrow(MessagesParsingException.class);
+
+        final OutboundMeshMessage outboundMeshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY_ACK);
+        when(outboundMeshMessageBuilder.buildNhsAck(eq(WorkflowId.PATHOLOGY), isA(MessagesParsingException.class)))
+            .thenReturn(outboundMeshMessage);
```

Comment:
Minor usage note: `isA(Class)` was meant for verifications; `any(Class)` for training
```suggestion
        when(outboundMeshMessageBuilder.buildNhsAck(eq(WorkflowId.PATHOLOGY), any(MessagesParsingException.class)))
            .thenReturn(outboundMeshMessage);
```
but I think this still works